src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractCreateDatasetCommand.java

```diff
@@ -0,0 +1,168 @@
+ *
+ * @author michael
+ */
+@RequiredPermissions(Permission.AddDataset)
+public abstract class AbstractCreateDatasetCommand extends AbstractDatasetCommand<Dataset> {
+    
+    private static final Logger logger = Logger.getLogger(AbstractCreateDatasetCommand.class.getCanonicalName());
+    
+    final protected boolean registrationRequired;
+    
+    public AbstractCreateDatasetCommand(Dataset theDataset, DataverseRequest aRequest) {
+        this(theDataset, aRequest, false);
+    }
+
+    public AbstractCreateDatasetCommand(Dataset theDataset, DataverseRequest aRequest, boolean isRegistrationRequired) {
+        super(aRequest, theDataset);
+        registrationRequired = isRegistrationRequired;
+    }
+   
+    protected void additionalParameterTests(CommandContext ctxt) throws CommandException {
+        // base class - do nothing.
+    }
+    
+    protected DatasetVersion getVersionToPersist( Dataset theDataset ) {
+        return theDataset.getLatestVersion();
+    }
+    
+    /**
+     * Called after the dataset has been persisted, but before the persistence context
+     * has been flushed. 
+     * @param theDataset The em-managed dataset.
+     * @param ctxt 
+     * @throws edu.harvard.iq.dataverse.engine.command.exception.CommandException 
+     */
+    protected void postPersist( Dataset theDataset, CommandContext ctxt ) throws CommandException {
+        // base class - default to nothing.
+    }
+    
+    protected abstract void handlePid( Dataset theDataset, CommandContext ctxt ) throws CommandException ;
+    
+    @Override
+    public Dataset execute(CommandContext ctxt) throws CommandException {
+        
+        additionalParameterTests(ctxt);
+        
+        Dataset theDataset = getDataset();
+        GlobalIdServiceBean idServiceBean = GlobalIdServiceBean.getBean(ctxt);
+        if ( isEmpty(theDataset.getIdentifier()) ) {
+            theDataset.setIdentifier(ctxt.datasets().generateDatasetIdentifier(theDataset, idServiceBean));
+        }
+        
+        DatasetVersion dsv = getVersionToPersist(theDataset);
+        // This re-uses the state setup logic of CreateDatasetVersionCommand, but
+        // without persisting the new version, or altering its files. 
+        new CreateDatasetVersionCommand(getRequest(), theDataset, dsv).prepareDatasetAndVersion();
+        
+        theDataset.setCreator((AuthenticatedUser) getRequest().getUser());
+        
+        theDataset.setCreateDate(getTimestamp());
+
+        theDataset.setModificationTime(getTimestamp());
+        for (DataFile dataFile: theDataset.getFiles() ){
+            dataFile.setCreator((AuthenticatedUser) getRequest().getUser());
+            dataFile.setCreateDate(theDataset.getCreateDate());
+        }
+        
+        String nonNullDefaultIfKeyNotFound = "";
+        if (theDataset.getProtocol()==null) theDataset.setProtocol(ctxt.settings().getValueForKey(SettingsServiceBean.Key.Protocol, nonNullDefaultIfKeyNotFound));
```

Comment:
indentation and brakcets please

Reply:
Done. (we might need to edit the coding guidelines, but agreed that for now we need indentation and brackets).